File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays(self, nums: list[int], k: int) -> int:
        n = len(nums)
        res = 0
        sum_ = 0
        i = 0

        for j in range(n):
            sum_ += nums[j]
            while i <= j and sum_ * (j - i + 1) >= k:
                sum_ -= nums[i]
                i += 1
            res += (j - i + 1)
        
        return res

File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) {
        int n = nums.size();
        long long res=0,sum=0;
        int i=0,j=0;
        while(j<n){
            sum+=nums[j];
            while(i<=j and sum*(j-i+1)>=k){
                sum-=nums[i];
                i++;
            }
            res+=(j-i+1);
            j++;
        }
        return res;
    }
};